statsmodels/survey/summary_stats.py

```diff
@@ -0,0 +1,538 @@
+            boot, etc
+
+        Returns
+        -------
+        An array containing the statistic calculated on the columns
+        of the dataset.
+        """
+        return np.dot(weights, self.data)
+
+
+class SurveyQuantile(SurveyStat):
+    """
+    Calculates the quantiles[s] for each column.
+
+    Parameters
+    -------
+    design : SurveyDesign object
+    data : ndarray
+        nxp array of the data to calculate the mean on
+    parameter: array-like
+        array of quantiles to calculate for each column
+
+    Attributes
+    ----------
+    data : ndarray
+        The data which to calculate the quantiles on
+    design :
+        Points to the SurveyDesign object
+    est : ndarray
+        The point estimates of the statistic, calculated on the columns
+        of data.
+    quantile : ndarray
+        The quantile[s] to calculate for each column
+    vc : ndarray
+        The variance-covariance of the estimates.
+    pseudo : ndarray
+        The jackknife pseudo-values.
+    """
+
+    def __init__(self, design, data, quantile, se_method, mse=False):
+        self.data = np.asarray(data)
+        super().__init__(design, mse)
+        self.quantile = np.asarray(quantile)
+
+        # give warning if user entered in quantile bigger than one
+        large_q = np.asarray([x > 1 for x in self.quantile])
+        n = large_q.sum()
+        if n > 0:
+            print("warning:", n, "inputed quantile[s] > 1")
+        self.n_cw = len(self.design.weights)
+
+        # get quantile[s] for each column
+        self.est = [0] * self.data.shape[1]
+        # need to call this several times
+        self.vc = [0] * self.data.shape[1]
+        if se_method == "jack":
+            for index in range(self.data.shape[1]):
+                self.est[index], self.vc[index] = self.jack(index)
+        elif se_method == "boot":
+            for index in range(self.data.shape[1]):
+                self.est[index], self.vc[index] = self.bootstrap(index)
+        else:
+            raise ValueError("method %s not supported" % se_method)
+    def _stat(self, weights, col_index):
+        quant_list = []
+        cw = np.cumsum(weights)
+        sorted_data = np.sort(self.data[:, col_index])
+        q = self.quantile.copy() * cw[-1]
+        # find index i such that self.cumsum_weights[i] >= q
+        ind = np.searchsorted(cw, q)
+
+        for i, pos in enumerate(ind):
+            # if searchsorted returns length of list
+            # return last observation
+            if pos in np.array([self.n_cw - 1, self.n_cw]):
+                quant_list.append(sorted_data[-1])
+                continue
+            if (cw[pos] == q[i]):
+                quant_list.append((sorted_data[pos] + sorted_data[pos+1]) / 2)
+            else:
+                quant_list.append(sorted_data[pos])
+        return quant_list
+
+
+class SurveyMedian(SurveyQuantile):
+    """
+    Wrapper function that calls SurveyQuantile
```

Comment:
Confusing docstring, this is a derived class, not a wrapper function.  But the user doesn't need to know this.  Just tell them what it is, e.g. `Calculate median for complex survey data.`.